packages/flutter/test/material/bottom_navigation_bar_test.dart

```diff
@@ -782,6 +782,57 @@ void main() {
     expect(tester.widget<Material>(backgroundMaterial).color, Colors.green);
   });
 
+  testWidgets('BottomNavigationBar shifting backgroundColor with transition', (WidgetTester tester) async {
+    // Regression test for: https://github.com/flutter/flutter/issues/22226
+
+    int _currentIndex = 0;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return Scaffold(
+              bottomNavigationBar: RepaintBoundary(
+                child: BottomNavigationBar(
+                  type: BottomNavigationBarType.shifting,
+                  currentIndex: _currentIndex,
+                  onTap: (int index) {
+                    setState(() {
+                      _currentIndex = index;
+                    });
+                  },
+                  items: const <BottomNavigationBarItem>[
+                    BottomNavigationBarItem(
+                      title: Text('Red'),
+                      backgroundColor: Colors.red,
+                      icon: Icon(Icons.dashboard),
+                    ),
+                    BottomNavigationBarItem(
+                      title: Text('Green'),
+                      backgroundColor: Colors.green,
+                      icon: Icon(Icons.menu),
+                    ),
+                  ],
+                ),
+              ),
+            );
+          },
+        ),
+      ),
+    );
+
+    await tester.tap(find.text('Green'));
+
+    for (int pump = 0; pump < 8; pump++) {
+      await tester.pump(const Duration(milliseconds: 30));
```

Comment:
afaik you shouldn't need the duration in this call to `pump()` - does the test fail without it?

Reply:
So `.pumpAndSettle()` has a default duration of 100ms, but `.pump()` doesn't have a default duration. Whenever I did it without defining a duration, all the goldens generated were the same image, and they were all the first frame. So it seems that without a duration, the animation never occurs because it loops through them all instantly and never tests past the first frame.